api/server/server.go

```diff
@@ -384,8 +386,14 @@ func createRouter(s *Server) *mux.Router {
 			localFct := fct
 			localMethod := method
 
+			// Define the context that we'll pass around to share info
+			// like the requestID.
+			reqID := stringid.TruncateID(stringid.GenerateNonCryptoID())
+			ctx := context.Background()
+			ctx = context.WithValue(ctx, "requestID", reqID)
```

Comment:
Eventually we’ll need to move that info into a non-http-specific data object (since the low-level code won’t deal with, or have access to, the http bits) and I think doing that as soon as possible in the flow is best so that we have a consistent pattern for access this “request specific data” irrespective of where we are in the code.  For example, I think looking for it in HTTP headers in one part of the code and then in a context object in other parts could lead to confusion, or worse, the data getting out of sync. 

-Doug


> On Sep 2, 2015, at 9:25 AM, David Calavera <notifications@github.com> wrote:
> 
> In api/server/server.go <https://github.com/docker/docker/pull/15953#discussion_r38530521>:
> 
> > @@ -384,8 +386,14 @@ func createRouter(s *Server) *mux.Router {
> >  			localFct := fct
> >  			localMethod := method
> >  
> > +			// Define the context that we'll pass around to share info
> > +			// like the requestID.
> > +			reqID := stringid.TruncateID(stringid.GenerateNonCryptoID())
> > +			ctx := context.Background()
> > +			ctx = context.WithValue(ctx, "requestID", reqID)
> Now that I think about this ID in the context, if we add it to the request header, we don't need to change every handler signature.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/docker/pull/15953/files#r38530521>.
>